packages/curvenote/src/check/examples/abstractLength.ts

```diff
@@ -0,0 +1,42 @@
+import path from 'node:path';
+import { selectFile } from 'myst-cli';
+import { TemplateOptionType, copyNode, extractPart, toText } from 'myst-common';
+import type { ISession } from '../../session/types.js';
+import { CheckStatus } from '../types.js';
+import type { Check, CheckInterface } from '../types.js';
+
+export const abstractLength: CheckInterface = {
+  id: 'abstract-length',
+  title: 'Abstract Length',
+  description: 'Ensure abstract in MyST project is the correct length',
+  category: 'abstract',
+  options: [
+    {
+      id: 'max',
+      title: 'Maximum word count for abstract',
+      required: true,
+      type: TemplateOptionType.string,
+    },
+  ],
+  validate: async (session: ISession, file: string, options: Check) => {
+    const { mdast } = selectFile(session, path.resolve(file)) ?? {};
+    if (!mdast) {
+      return { status: CheckStatus.error, message: `Error loading content from ${file}` };
+    }
+    const abstract = extractPart(copyNode(mdast), 'abstract');
+    if (!abstract) {
+      return { status: CheckStatus.error, message: `No abstract found in ${file}` };
+    }
+    const splitAbstract = toText(abstract).split(/\s+/);
```

Comment:
We have used this package before:
https://www.npmjs.com/package/@wordpress/wordcount

Reply:
Probably the same as default though.. !
https://github.com/WordPress/gutenberg/blob/f67c38677442518e65e338849eaa3fc435773a5c/packages/wordcount/src/defaultSettings.js#L96